client/conn.go

```diff
@@ -103,6 +98,12 @@ func ConnectWithDialer(ctx context.Context, network string, addr string, user st
 		options[i](c)
 	}
 
+	if c.tlsConfig != nil {
+		seq := c.Conn.Sequence
+		c.Conn = packet.NewTLSConn(conn)
+		c.Conn.Sequence = seq
```

Comment:
`packet.NewConn(conn)` just created some buffer, it didn't write MySQL data so I think Sequence is untouched.

Reply:
Yet it could be modified via an option call since it's a public field